app/controllers/api/locations_controller.rb

```diff
@@ -24,12 +27,22 @@ def current_resource
     Location.find_by_code(params[:barcode]) if params[:barcode]
   end
 
-  def process_location
-    if @location.submit(params)
+  def process_location(succeeded)
+    if succeeded
       render json: @location, serializer: "#{@location.type}Serializer".constantize
     else
       render json: { errors: @location.errors.full_messages}, status: :unprocessable_entity
     end
   end
 
-end
\ No newline at end of file
+  def permitted_params
```

Comment:
This method is the same found in  `LocationController`. It would be good to factor it out (but I'm not sure what's the best way to do it).